OrbitGl/Batcher.cpp

```diff
@@ -220,17 +225,28 @@ void Batcher::GetBoxGradientColors(const Color& color, std::array<Color, 4>* col
 }
 
 void Batcher::ResetElements() {
-  line_buffer_.Reset();
-  box_buffer_.Reset();
-  triangle_buffer_.Reset();
+  for (auto& [_, buffer] : buffers_by_layer_) {
+    buffer.line_buffer.Reset();
+    buffer.box_buffer.Reset();
+    buffer.triangle_buffer.Reset();
+  }
 }
 
 void Batcher::StartNewFrame() {
   ResetElements();
   user_data_.clear();
 }
 
-void Batcher::Draw(bool picking) const {
+std::set<float> Batcher::GetLayers() const {
```

Comment:
Why does this return a set? Do you need this to be a set? If not, use a vector (which should be the default choice unless you need something different).

Reply:
Because we don't want to have repeated layers, but we can use a vector with sort and unique. I will change to that.